cpp/src/copying/slice.cpp

```diff
@@ -19,28 +19,53 @@
 #include <cudf/column/column_view.hpp>
 #include <cudf/copying.hpp>
 #include <cudf/detail/copy.hpp>
+#include <cudf/detail/null_mask.hpp>
+
 #include <algorithm>
 
 namespace cudf {
-
 namespace experimental {
 
-std::vector<cudf::column_view> slice(cudf::column_view const& input,
-                                                std::vector<size_type> const& indices){
+namespace detail {
 
-    CUDF_EXPECTS(indices.size()%2 == 0, "indices size must be even");
-    std::vector<cudf::column_view> result{};
+std::vector<column_view> slice(column_view const& input,
+                               std::vector<size_type> const& indices,
+                               cudaStream_t stream){
+  CUDF_EXPECTS(indices.size() % 2 == 0, "indices size must be even");
 
-    if(indices.size() == 0 or input.size() == 0) {
-        return result;
-    }
+  std::vector<column_view> result{};
 
-    for(size_t i = 0; i < indices.size(); i+=2){
-        result.emplace_back(detail::slice(input, indices[i], indices[i+1]));
+  if(indices.size() == 0 or input.size() == 0) {
+    return result;
+  }
+
+  auto null_counts = cudf::detail::segmented_count_unset_bits(input.null_mask(),
+                                                              indices, stream);
+
+  for (size_t i = 0; i < indices.size() / 2; i++) {
+    std::vector<column_view> children{};
```

Comment:
Probably better to move this out of the loop and copy to result instead of move:
- Replaces the repeated initialization of children with repeated copies (faster);
- Clearer that the children array is the same in every iteration;